src/Maestro/Maestro.Contracts/MergePolicyEvalutationResult.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Maestro.Contracts
+{
+    public class MergePolicyEvaluationResults
+    {
+        public MergePolicyEvaluationResults(IEnumerable<MergePolicyEvaluationResult> results)
+        {
+            Results = results.ToImmutableList();
+        }
+
+        public IReadOnlyList<MergePolicyEvaluationResult> Results { get; }
+
+        public bool Succeeded => Results.Count > 0 && Results.All(r => r.Status == MergePolicyEvaluationStatus.Success);
+
+        public bool Pending => Results.Count > 0 && Results.Any(r => r.Status == MergePolicyEvaluationStatus.Pending);
```

Comment:
Any already returns false if the thing is empty, so the "Results.Count" is unnecessary. (Weirdly the same is _not_ true of All, so it is required there)